src/integer_mod_q/mat_zq/get.rs

```diff
@@ -144,7 +145,100 @@ impl GetEntry<Zq> for MatZq {
 }
 
 impl MatZq {
-    #[allow(dead_code)]
+    /// Outputs the row vector of the specified row.
+    ///
+    /// Parameters:
+    /// - `row`: specifies the row of the matrix
+    ///
+    /// Returns a row vector of the matrix at the position of the given
+    /// row or an error, if the number of rows is
+    /// greater than the matrix or negative.
+    ///
+    /// # Example
+    /// ```rust
+    /// use qfall_math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let matrix = MatZq::from_str("[[1, 2, 3],[3, 4, 5]] mod 4").unwrap();
+    ///
+    /// let row0 = matrix.get_row(0).unwrap(); // first row
+    /// let row1 = matrix.get_row(1).unwrap(); // second row
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if the number of the row is greater than the matrix or negative.
+    pub fn get_row(&self, row: impl TryInto<i64> + Display + Copy) -> Result<Self, MathError> {
+        let row_i64 = evaluate_index(row)?;
+
+        if self.get_num_rows() <= row_i64 {
+            return Err(MathError::OutOfBounds(
+                format!("be smaller than {}", self.get_num_rows()),
+                format!("{}", row_i64),
+            ));
+        }
+
+        let out = MatZq::new(1, self.get_num_columns(), self.get_mod()).unwrap();
+        for column in 0..self.get_num_columns() {
+            unsafe {
+                fmpz_set(
+                    fmpz_mat_entry(&out.matrix.mat[0], 0, column),
+                    fmpz_mod_mat_entry(&self.matrix, row_i64, column),
+                )
+            };
+        }
+        Ok(out)
+    }
+
+    /// Outputs a column vector of the specified column.
+    ///
+    /// Input parameters:
+    /// * `column`: specifies the column of the matrix
+    ///
+    /// Returns a column vector of the matrix at the position of the given
+    /// column or an error, if the number of columns is
+    /// greater than the matrix or negative.
+    ///
+    /// # Example
+    /// ```rust
+    /// use qfall_math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let matrix = MatZq::from_str("[[1, 2, 3],[3, 4, 5]] mod 4").unwrap();
+    ///
+    /// let col0 = matrix.get_column(0).unwrap(); // first column
+    /// let col1 = matrix.get_column(1).unwrap(); // second column
+    /// let col2 = matrix.get_column(2).unwrap(); // third column
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if the number of the column is greater than the matrix or negative.
+    pub fn get_column(
+        &self,
+        column: impl TryInto<i64> + Display + Copy,
+    ) -> Result<Self, MathError> {
```

Comment:
Probably add a warning that if the matrix is reduced, the column vector will also be reduced. Otherwise, it won't, i.e. it relies on the the initial matrix to be reduced. - Same for `get_row`.

Reply:
Yes, we get reduced matrices from all `from` methods, but that does not guarantee that every `MatZq` is always reduced. It's just a comment (for developers in the code) to find issues easier.